rfcs/text/0000-extend-file-pe.md

```diff
@@ -0,0 +1,142 @@
+# 0000: Extend the PE field set
+
+- Stage: **1 (proposal)**
+- Date: **TBD**
+
+The Portable Executable (PE) sub-field, of the `file` top-level fieldset, can be updated to include more file attributes to aid in file analysis. This additional document metadata can be used for malware research, as well as coding and other application development efforts.
+
+## Fields
+
+This RFC is to create 25 additional sub-fields within the `file.pe` fieldset.
+
+| Name | Type | Description |
+| ---- | ---- | ----------- |
+| file.pe.authentihash | keyword | Authentihash of the PE file. |
+| file.pe.compile_timestamp | date | Compile timestamp of the PE file. |
+| file.pe.compiler | keyword | Compiler information. |
+| file.pe.compiler.version | keyword | Version of the compiler. |
+| file.pe.compiler.name | keyword | Name of the compiler. |
+| file.pe.creation_date | date | Extracted when possible from the file's metadata. Indicates when it was built or compiled. It can also be faked by malware creators. |
+| file.pe.entry_point | keyword | Entry point of the PE file. |
+| file.pe.exports | keyword | List of symbols exported by PE |
+| file.pe.debug | nested | Debug information, if present |
+| file.pe.debug.offset | keyword | Debug offset information. |
+| file.pe.debug.size | keyword | Size of the debug information. |
+| file.pe.debug.type_str | keyword | Format of the debug information. |
+| file.pe.debug.type | keyword | Information type generated by the debug options. |
+| file.pe.debug.timestamp | date | Timestamp of the debug information. |
+| file.pe.imports | flattened | List of all imported functions |
```

Comment:
would a multi-valued keyword make sense for this? windebug notation? like `advapi32!RegSetValueExW`

Reply:
That seems kind of janky to query, no? Assuming you wanted to query individually by all binaries that imported a particular function or imported a particular DLL

Reply:
Is there a recommended change?